src/NServiceBus.Core/Pipeline/Outgoing/OutgoingPublishContext.cs

```diff
@@ -1,29 +1,30 @@
 ﻿namespace NServiceBus.OutgoingPipeline
 {
     using NServiceBus.Pipeline;
+    using PublishOptions = NServiceBus.PublishOptions;
 
     /// <summary>
     /// Pipeline context for publish operations.
     /// </summary>
-    public class OutgoingPublishContext : BehaviorContext
+    public class OutgoingPublishContext : OutgoingContext
     {
         /// <summary>
-        /// The message to be published.
-        /// </summary>
-        public OutgoingLogicalMessage Message { get; private set; }
-
-        /// <summary>
         /// Initializes the context with a parent context.
         /// </summary>
         public OutgoingPublishContext(OutgoingLogicalMessage message, PublishOptions options, BehaviorContext parentContext)
             : base(parentContext)
         {
             Message = message;
-            Guard.AgainstNull("parentContext", parentContext);
-            Guard.AgainstNull("message", message);
-            Guard.AgainstNull("options", options);
+            Guard.AgainstNull(nameof(parentContext), parentContext); // TODO DanielTim : Is this needed here
```

Comment:
it doesn't. passing null is used to mark a context as the root context (and is currently only done by RootContext).